docs/develop/java/workflow_signals.mdx

```diff
@@ -0,0 +1,48 @@
+---
+sidebar_position: 7
+description: "Find out how Restate services can send requests to each other."
+---
+
+import Admonition from '@theme/Admonition';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Workflow Signals
+
+**This feature only works for services of the Workflow type.**
+
+You can use signals to interact with your running workflows.
+
+You can use them to let the workflow block until an event occurs, or to send a signal / information to a running workflow.
+
+You can implement signals via the Durable Promises feature of the SDK.
+These promises are durable and distributed, meaning they survive crashes and can be resolved or rejected by any handler in the workflow.
+
+Do the following:
+1. Create a promise in your `run` handler that is durable and distributed
+2. Resolve or reject the promise in any other handler in the workflow.
```

Comment:
It is very important to say that this can be completed only once!!!

Reply:
Isn't that always the case for promises?